test/models/tagging_test.rb

```diff
@@ -1,73 +1,78 @@
 require "test_helper"
 
 class TaggingTest < ActiveSupport::TestCase
-  test "tagging a hackathon by name" do
-    hackathon = hackathons(:assemble)
+  setup do
+    @hackathon = hackathons(:assemble)
+  end
+
+  test "checking for tags" do
+    assert @hackathon.tags.none?
+
+    @hackathon.taggings.create! tag: Tag.new(name: "Tag")
+
+    assert @hackathon.tagged_with? "Tag"
+    assert @hackathon.tagged_with? Tag.last
+
+    assert_not @hackathon.tagged_with? "Nonexistent"
```

Comment:
```suggestion
    assert_not @hackathon.tagged_with? "Nonexistent"
    assert_not @hackathon.tagged_with? nil
```

Reply:
What would this do?

Reply:
We should test of edge cases, such as passing `nil` to `tagged_with?` should returns false